exotations/task_maps/task_map/src/JointJerkBackwardDifference.cpp

```diff
@@ -0,0 +1,116 @@
+ *  * Neither the name of  nor the names of its contributors may be used to
+ *    endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+#include "task_map/JointJerkBackwardDifference.h"
+
+REGISTER_TASKMAP_TYPE("JointJerkBackwardDifference", exotica::JointJerkBackwardDifference);
+
+namespace exotica
+{
+constexpr double backward_difference_params_[3] = {-3, 3, -1};  //!< Binomial cooeficient parameters.
+
+JointJerkBackwardDifference::JointJerkBackwardDifference() = default;
+JointJerkBackwardDifference::~JointJerkBackwardDifference() = default;
+
+void JointJerkBackwardDifference::assignScene(Scene_ptr scene)
+{
+    scene_ = scene;
+
+    // Get ndof
+    N_ = scene_->getSolver().getNumControlledJoints();
+
+    // Check start state is correct length
+    if (init_.StartState.size() != N_) throw_named("StartState is wrong size!");
+
+    // Init each col of q_ with start state
+    q_.resize(N_, 3);
+    for (int i = 0; i < 3; i++)
+        q_.col(i) = init_.StartState;
+
+    // Init identity matrix
+    I_ = Eigen::MatrixXd::Identity(N_, N_);
+}
+
+void JointJerkBackwardDifference::Instantiate(JointJerkBackwardDifferenceInitializer& init)
+{
+    init_ = init;
+}
+
+void JointJerkBackwardDifference::setPrevJointState(Eigen::VectorXdRefConst joint_state)
+{
+    /** \brief Logs previous joint state.
+   *
+   *  setPrevJointState must be called after solve is called in a Python/C++ script is called 
+   *  to ensure the time-derivatives are appropriately approximated. 
+   *  Each previous joint state is pushed back by collumn and the given joint_state is placed
+   *  in q_.col(0). 
+   *  Finally, we compute the new qbd_.
+   * 
+   */
+
+    // Input check
+    if (joint_state.size() != N_) throw_named("Wrong size for joint_state!");
+
+    // Push back previous joint states
+    q_.col(2) = q_.col(1);
+    q_.col(1) = q_.col(0);
+    q_.col(0) = joint_state;
+
+    // Compute new qbd_
+    qbd_ = q_.col(0) * backward_difference_params_[0] + q_.col(1) * backward_difference_params_[1] + q_.col(2) * backward_difference_params_[2];
```

Comment:
I really liked the previous solution with the Vector3d in this case - because with the right settings a Nx3 matrix times a 3x1 vector can be parallelized automatically with alignment. So while this works there is a tiny performance penalty.

Reply:
If `backward_difference_params_` is reset to a `Vector3d` then I don't know how to make it a `constexpr`. I tried 
```
constexpr Eigen::Vector3d backward_difference_params_ << -3, 3, -1;
```
but this throws an error at compile-time. It doesn't like the `<<` operator.
Is your suggestion to revert back to non-constant `Vector3d`?

Reply:
@VladimirIvan unfortunately both these suggestions fail to compile.

Reply:
Never mind, eigen doesn't support constexpr like this (at least not until higher versions of eigen and c++ standard if I read the docs correctly)